public/static/docs/api-reference/get_url.md

```diff
@@ -0,0 +1,76 @@
+# dvc.api.get_url()
+
+Returns the full URL
+[string](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str)
+to the physical location (in a [DVC remote](/doc/command-reference/remote)) of a
+<abbr>data artifact</abbr> specified by its `path` in a `repo` (<abbr>DVC
+project</abbr>).
+
+⚠️ Note that the returned URL is formed by analyzing the corresponding
+[DVC-file](/doc/user-guide/dvc-file-format) (see [Examples](#examples) below).
+**There is no guarantee that the file actually exists in that location**. Please
+keep this in mind when using the URL string in your code.
+
+💡 Having the resource's URL, it should be possible to download it directly with
+an appropriate tool such as
+[`urlretrieve`](https://docs.python.org/3/library/urllib.request.html#urllib.request.urlretrieve)
+or `boto3`
+[download_fileobj](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Object.download_fileobj).
+
+> For possible location protocols, refer to the
+> [supported remote types](https://dvc.org/doc/command-reference/remote/add#supported-storage-types)
+
+## Signature
+
+```py
+get_url(path, repo=None, rev=None, remote=None)
+```
+
+## Parameters
+
+- **`path`** - used to specify the location of the target artifact within the
+  source project in `repo`, relative to the project's root.
+
+- `repo` - specifies the location of the source DVC project. Both HTTP and SSH
+  protocols are supported for online Git repository URLs (e.g.
+  `[user@]server:project.git`). `repo` can also be a local file system path to
+  an "offline" project. If not supplied, this defaults to the current working
+  directory.
+
+  > A `NotDvcRepoError` is thrown if `repo` is not a valid DVC project.
```

Comment:
Are the exceptions going to be documented? It'd be good to mention of where the exceptions exist so they could import.

Reply:
Please make that in different PR. I just wanted to know the what the plans for *exceptions* are.

Reply:
Good question. I don't have a specific plan for documenting exceptions. Probably not? They can just be mentioned in the context of each function.

I added the full module paths to the exceptions mentioned now. They all come from `dvc.exceptions` anyway but it's a good idea. Please check, thanks.

Reply:
p.s. noted about exceptions in https://github.com/iterative/dvc/issues/3176#issuecomment-576539623

Reply:
> They all come from dvc.exceptions anyway but it's a good idea.

This is not always the case, at least with `get_url()` and `summon()` (this doc does not cover `summon`, right?). 

https://github.com/iterative/dvc/blob/04161ad6f2d24b02798aa79fe2fcd8fbb76dab15/dvc/api.py#L46

This is a recent change, though. And, that's why I asked what would be the case with documentation of the exceptions? Exceptions are part of the API.